test/function.test.ts

```diff
@@ -0,0 +1,118 @@
+import { runInterpreter } from '../src/interpreter/runner.js';
+import { expect, test } from 'vitest';
+
+test('identity function', async() => {
+  const input = `
+    fun returnSum(a: number, b: number): number {
+      print "returnSum called";
+      return a + b;
+    }
+
+    fun identity(a: (number, number) => number): (number, number) => number {
+      print "identity called";
+      return a;
+    }
+
+    print identity(returnSum)(27, 15); // prints "42";
+  `;
+
+  const expectedOutput = `
+    identity called
+    returnSum called
+    42
+  `;
+
+  await runInterpreterAndAssertOutput(input, expectedOutput);
+});
+
+test('pass reference to function and call it', async() => {
+  const input = `
+    fun aFunction(aLambda: (number) => number, aNumber: number): number {
+        print "aFunction called";
+        return aLambda(aNumber);
+    }
+
+    fun aTimesTwo(a: number): number {
+        print "aTimeTwo called";
+        return a * 2;
+    }
+
+    var result = aFunction(aTimesTwo, 9);
+    print result;
+  `;
+
+  const expectedOutput = `
+    aFunction called
+    aTimeTwo called
+    18
+  `;
+
+  await runInterpreterAndAssertOutput(input, expectedOutput);
+});
+
+test('Closure 1', async() => {
+  const input = `
+    // So far fails with: No variable 'outside' defined
```

Comment:
maybe failing tests should be commented out and marked with a TODO or FIXME comment?